### ISSUE TITLE
IDISP001 is being triggered in .net 6 style asp.net program/startup classes

### ISSUE THREAD

**Issue by xt0rted** [DotNetAnalyzers/IDisposableAnalyzers#330]
After migrating my application to the new top-level statement program/startup format in asp.net 6 `IDISP001` is being triggered on the following line:

```csharp
var app = builder.Build();
```

This can be seen by creating a new asp.net core web app, installing this analyzer & setting the rule to error, and building.

If I disable this rule then I get a second error of [IDE0079](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0079) for the whole code block.

```csharp
#pragma warning disable IDISP001 // Dispose created.
var app = builder.Build();
#pragma warning restore IDISP001 // Dispose created.
```

**Comment by JohanLarsson** [comment]
Why not:
```cs
using var app = builder.Build();
```
